src/components/WorkspaceDetails/WorkspaceDetails.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import {
+  OverviewPage,
+  DetailsPage,
+  DetailsItem,
+  k8sGetResource,
+  K8sModelCommon /* , useK8sWatchResource */,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import type { K8sResourceCommon } from '@openshift/dynamic-plugin-sdk-utils';
+import { useParams } from 'react-router-dom';
+
+const WorkspaceDetailsTabs = (workspace: K8sResourceCommon, loaded: boolean, error: { message: string; status: number }) => {
+  return [
+    {
+      key: 'overview',
+      title: 'Overview',
+      content: (
+        <OverviewPage loaded={loaded} resource={workspace} loadError={error} rightColumn={<></>}>
+          <DetailsItem label="Services" resource={workspace} path="spec.type.name" loaded={loaded} />
+        </OverviewPage>
+      ),
+      ariaLabel: 'Overview',
+    },
+  ];
+};
+
+const WorkspaceDetailsBreadcrumbs = (workspaceName: string) => [
+  { name: 'Workspaces', path: '/workspaces' },
+  { name: 'Workspace details', path: workspaceName ? `/workspaces/${workspaceName}` : '' },
+];
+
+const WorkspaceModel: K8sModelCommon = {
+  apiVersion: 'v1beta1',
+  apiGroup: 'tenancy.kcp.dev',
+  kind: 'Workspace',
+  plural: 'workspaces',
+};
+
+const WorkspaceDetails = () => {
+  const { workspaceName } = useParams();
+
+  const [workspace, setWorkspace] = React.useState<K8sResourceCommon | undefined>();
+  const [loaded, setLoaded] = React.useState(false);
+  const [error, setError] = React.useState<
+    | {
+        message: string;
+        status: number;
+      }
+    | undefined
+  >();
+
+  // Watch isn't currently working
+  // const watchedResource = {
+  //   isList: false,
+  //   groupVersionKind: {
+  //     group: 'tenancy.kcp.dev',
+  //     version: 'v1beta1',
+  //     kind: 'Workspace',
+  //   },
+  // };
+
+  // console.log('KKD - workspaceName', workspaceName);
+  // const [workspaceWatch, loadedWatch, errorWatch] = useK8sWatchResource(watchedResource, {
+  //   wsPrefix: `ws/kcp/${workspaceName}`,
+  //   pathPrefix: `/api/kcp/${workspaceName}`,
+  // });
+  // console.log('KKD - workspace:', workspaceWatch, loadedWatch);
+
+  React.useEffect(() => {
+    setLoaded(false);
+    setError(undefined);
+    k8sGetResource({
+      model: WorkspaceModel,
+      queryOptions: { path: workspaceName },
+    })
+      .then((response) => {
+        setWorkspace(response as K8sResourceCommon);
+      })
+      .catch((err) => {
+        setError({ message: err.message, status: err.json.code });
+      })
+      .finally(() => {
+        setLoaded(true);
+      });
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <div style={{ overflow: 'scroll' }}>
```

Comment:
External style since we should avoid inline styles and we're likely to need this in other pages?

Reply:
Ha - it is used on other pages, so I copied what was used there.  Instead of using an external CSS, how do we feel about incorporating styled-components?  A find it to be a much cleaner way to incorporate special case CSS.

Reply:
A styled component makes sense for reuse, but I still think the associated styles should go in a separate scss file.

Reply:
I would say take a look at what I changed.  It is easier to tie specific css to a particular component so you have control when the css is applied.  Secondly styled components only load the css the first time it is used.  So it isn't called if the components is never called and it won't call the css more than once if has already been used.